### PR TITLE
docs: 0.13.1 fee updates

### PR DESCRIPTION
**Pull request by ArielElp**
### Description of the Changes

Starknet v0.13.1 allows state diff to be sent to L1 via blobs instead of calldata. This change should be reflected in the final fee formula.

Additionally, L2 calldata, events, and code are now priced. These new prices should also appear in the fee mechanism page.

### PR Preview URL

https://starknet-io.github.io/starknet-docs/pr-1162/documentation/architecture_and_concepts/Network_Architecture/fee-mechanism/

### Check List

- [ ] Changes have been done against main branch, and PR does not conflict
- [X] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1162)

### CODE REVIEW

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc [hunk 1]
```diff
@@ -33,40 +33,73 @@ where:
...
 
-include::partial$snippet_v3_max_price_sidebar.adoc[]
+[#deprecated_fee_limitations]
+=== Deprecated transactions (verion < 3)
 
-For transaction versions before v3 transactions, the `max_fee` field specifies the maximum fee.
+With older transactio versions, users specify the maximum fee that they are willing to pay for a transaction.
 
-The only limitation on the sequencer, which is enforced by the Starknet OS, is that the actual fee charged is bounded by `max_fee`, but for now, StarkWare's sequencer only charges the fee required to cover the proof cost, which is potentially less than the maximum fee.
+The only limitation on the sequencer, which is enforced by the Starknet OS, is that the actual fee charged is bounded by `max_fee`. While not enforced in the proof, the Starknet sequencer usually charges less than `max_fee`, as it charges exactly the amount given by the above fee formula.
```
Comment (leo-starkware) [inline components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc:93]: "exactly the amount of the formula": syscalls don't appear though, so I'm not sure about the word "exactly"

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc [hunk 2]
```diff
@@ -33,40 +33,73 @@ where:
...
-Currently, the sequencer only takes into account L1 costs involving proof submission. The following two components affect the L1 footprint of a transaction:
+[#what_do_we_pay_for]
+== What do we price
 
-* xref:#computation_without_builtins[Computational complexity]: A transaction's portion of the proof verification cost is proportional to its complexity.
-* xref:#onchain_data_components[Onchain data]: L1 calldata cost originating from xref:Network_Architecture/on-chain-data.adoc[data availability] and L2→L1 messages.
+At the time of writing, the following components are contributing to the transacation fee:
+
+* xref:#computation_without_builtins[Computational complexity]: The marginal cost of verifying the transaction on L1, measured in L1 gas.
+* xref:#onchain_data_components[Onchain data]: The cost of posting  the state diffs induced by the transaction to L1 (for more details, see xref:Network_Architecture/on-chain-data.adoc[data availability]). This is measured in L1 gas or L1 data gas, depending on whether or not the L2 block in which the transaction was included uses blobs or calldata.
```
Comment (leo-starkware) [inline components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc:100]: should be: uses calldata or blobs.

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc [hunk 3]
```diff
@@ -33,40 +33,73 @@ where:
...
+* L2→L1 messages: Messages sent to L1 are eventually sent to the Starknet core contract by the sequencer, which leads to the sending L2 transaction having additional L1 gas costs.
```
Comment (leo-starkware) [inline components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc:101]: I propose: "sent to the Starknet core contract as L1 calldata by the sequencer; therefore L2 transaction that send L2->L1 messages incur an additional L1 gas cost."

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc [hunk 4]
```diff
@@ -33,40 +33,73 @@ where:
...
+& - \; \text{contract_update_discount}\cdot (n-1) - 240 \\
+& + \; \text{message_calldata_cost}\cdot\left(3t + \sum\limits_{i=1}^t q_i\right) \\
+& + \; \text{storage_write_cost}\cdot t \\
+& + \; \text{l2_payload_costs}\Bigg)
+\end{align}
+++++
+
+where:
+
+* stem:[$v, w, n, m, t, \ell, \text{da_calldata_cost}, \text{message_calldata_cost}, D, \text{l2_payload_costs}$] are defined in the same manner as the blob-based formula above.
```
Comment (leo-starkware) [inline components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc:71]: "as in"

**Review by ArielElp — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1162)* status: 0 of 3 files reviewed, 5 unresolved discussions (waiting on @leo-starkware)

___
*[`components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc` line 49 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/1162#gh-1514175124:-NsIcGDv97Vv4YbZIqDl:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/034c7a90dd78c8f18be7852078f9d7dcbba73a65/components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc#L49)):*
<details><summary><i>Previously, leo-starkware wrote…</i></summary><blockquote>

And same below
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc` line 71 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/1162#gh-1514197558:-NsIW8WE0Tb4Oc_hUIOK:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/034c7a90dd78c8f18be7852078f9d7dcbba73a65/components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc#L71)):*
<details><summary><i>Previously, leo-starkware wrote…</i></summary><blockquote>

"as in"
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc` line 93 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/1162#gh-1514185351:-NsIcO_zBcJjL2QcYc8T:b-bsty5i) ([raw file](https://github.com/starknet-io/starknet-docs/blob/034c7a90dd78c8f18be7852078f9d7dcbba73a65/components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc#L93)):*
<details><summary><i>Previously, leo-starkware wrote…</i></summary><blockquote>

"exactly the amount of the formula": syscalls don't appear though, so I'm not sure about the word "exactly"
</blockquote></details>

Changed the wording a bit (one can argue it's included in the step count, that's how we measure it today)
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc` line 100 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/1162#gh-1514187631:-NsIWT1U6PrOrd4tc9p1:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/034c7a90dd78c8f18be7852078f9d7dcbba73a65/components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc#L100)):*
<details><summary><i>Previously, leo-starkware wrote…</i></summary><blockquote>

should be: uses calldata or blobs.
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc` line 101 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/1162#gh-1514191793:-NsIcsSf7SmCYJeOILcR:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/034c7a90dd78c8f18be7852078f9d7dcbba73a65/components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc#L101)):*
<details><summary><i>Previously, leo-starkware wrote…</i></summary><blockquote>

I propose: "sent to the Starknet core contract as L1 calldata by the sequencer; therefore L2 transaction that send L2->L1 messages incur an additional L1 gas cost."
</blockquote></details>

Done.

**Review by leo-starkware — COMMENTED** [review]
line 267: storage_write_cost needs to be l1_storage_write_cost

**Pull request merged**